lib/class-wp-block-supports.php

```diff
@@ -207,14 +207,25 @@ function get_block_wrapper_attributes( $extra_attributes = array() ) {
  * @return array Block attributes.
  */
 function wp_block_supports_track_block_to_render( $args ) {
-	if ( null !== $args['render_callback'] ) {
+	if ( is_callable( $args['render_callback'] ) ) {
 		$block_render_callback   = $args['render_callback'];
 		$args['render_callback'] = function( $attributes, $content, $block ) use ( $block_render_callback ) {
-			$parent_block                       = WP_Block_Supports::$block_to_render;
-			WP_Block_Supports::$block_to_render = $block->parsed_block;
-			$result                             = $block_render_callback( $attributes, $content, $block );
-			WP_Block_Supports::$block_to_render = $parent_block;
-			return $result;
+			// Check for null for back compatibility with WP_Block_Type->render
+			// which is unused since the introduction of WP_Block class.
+			//
+			// See:
+			// - https://core.trac.wordpress.org/ticket/49927
+			// - commit 910de8f6890c87f93359c6f2edc6c27b9a3f3292 at wordpress-develop.
+			//
+			if ( null !== $block ) {
+				$parent_block                       = WP_Block_Supports::$block_to_render;
+				WP_Block_Supports::$block_to_render = $block->parsed_block;
+				$result                             = $block_render_callback( $attributes, $content, $block );
+				WP_Block_Supports::$block_to_render = $parent_block;
+				return $result;
+			} else {
+				return $block_render_callback( $attributes, $content );
```

Comment:
Minor stylistic point: I'd prefer the early-return method (which we tend to prefer in JS too) over the if-else pair. And that way we also avoid double negatives like `!== null`:

```php
if ( null === $block ) {
  return $block_render_callback( $attributes, $content );
}

$parent_block = …
```

Reply:
Extra advantage for early-return – now the main logic branch is at the top execution level and the error handling is an obvious detail at a deeper level.

P.S. Sorry for the drive-by comment… :)

Reply:
@nb: I still remember _Else considered harmful_. ;)